### PR TITLE
[ARM/AARCH64] BootCode Changes to enable the Alignment Check bit and disable the unaligned access bit

### PR DESCRIPTION
**Pull request by simpal01**
We need changes in both Arm/AArch64 bootcode in order for the -mno-unaligned-access to work.

We need to enable the Alignment Check bit in the System Control Register (SCTLR)  which is the Bit 1 (A) and it turn on alignment fault checking for data accesses . If an unaligned data access occurs, an alignment fault exception will be generated.

Also need to disable the Unaligned Access Enable(U) bit in the System Control Register (SCTLR). As a result, unaligned data accesses will generate an alignment fault exception.

### CODE REVIEW

File: picocrt/machine/arm/crt0.c [hunk 1]
```diff
@@ -280,6 +280,10 @@ _cstart(void)
                 uint32_t sctlr;
                 __asm__("mrc p15, 0, %0, c1, c0, 0" : "=r" (sctlr));
                 sctlr |= SCTLR_ICACHE | SCTLR_BRANCH_PRED | SCTLR_DATA_L2 | SCTLR_MMU;
+                #ifndef __ARM_FEATURE_UNALIGNED
+                    sctlr |= (1 << 1);
```
Comment (arichardson) [inline picocrt/machine/arm/crt0.c:284]: Maybe add a symbolic #define to explain which bit each line sets?

**Comment by keith-packard** [comment]
We should be able to test this by creating a special build that adds -mno-unaligned-access to CFLAGS, right? We're going to need that so this change can be tested.

**Comment by simpal01** [comment]
> We should be able to test this by creating a special build that adds -mno-unaligned-access to CFLAGS, right? We're going to need that so this change can be tested.

Yes. I have a patch (https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm/pull/523) to add the **Armv7a NO FP -mno-unaligned-access** library variants in 
 LLVM-embedded-toolchain-for-Arm. But you also need these llvm changes (https://github.com/llvm/llvm-project/pull/114782)  so that -mno-unaligned-access can be added to the multilib flags.

**Comment by pratlucas** [comment]
Hi @keith-packard ,
The test for `-mno-unaligned-access` on Arm has been added in #866. The aarch64 test, on the other hand, depends on the merging of #888, as we could only get a data abort to happen on the FVP models.
Do you feel the current level of testing is sufficient for this to be merged? Or would you prefer to wait for #888?

**Comment by keith-packard** [comment]
I just read through #888 and I think it's ready to be merged; do you have any reason I should delay that? If not, we can merge that and then use that to test this change?

**Comment by pratlucas** [comment]
Hi @keith-packard ,
I have just double-checked and it looks like these changes to the A and U bits of SCTLR register were not included in #888 .
These bits are required to enable data aborts for unaligned accesses. Can we re-open this PR?

**Comment by keith-packard** [comment]
Can you rebase these changes so they can be integrated?

**Comment by pratlucas** [comment]
I've created #895 with the up-to-date changes, as I don't have push permissions to the original PR branch in @simpal01 's fork.

**Comment by pratlucas** [comment]
Thanks @keith-packard ! Now that #895 is merged, I believe this PR can be closed.

**Pull request closed without merging**